pageserver/src/tenant.rs

```diff
@@ -2932,21 +2973,38 @@ impl Tenant {
         Ok(new_timeline)
     }
 
+    /// For unit tests, make this visible so that other modules can directly create timelines
+    #[cfg(test)]
+    pub(crate) async fn bootstrap_timeline_test(
+        &self,
+        timeline_id: TimelineId,
+        pg_version: u32,
+        load_existing_initdb: Option<TimelineId>,
+        ctx: &RequestContext,
+    ) -> anyhow::Result<Arc<Timeline>> {
+        let uninit_mark = self.create_timeline_uninit_mark(timeline_id).unwrap();
+        self.bootstrap_timeline(
+            timeline_id,
+            pg_version,
+            load_existing_initdb,
+            uninit_mark,
+            ctx,
+        )
+        .await
+    }
+
```

Comment:
Just out of interest, did you understand why are do these methods exist at all compared to just using `create_timeline`

Reply:
Nope, didn't look into it.